### ISSUE TITLE
Default feature length 1024?

### ISSUE THREAD

**Issue by sailfish009** [agemagician/ProtTrans#21]
The default feature length seems to be 1024, 
is there any way to reduce the length of this feature to 64?

Thanks in advance.

**Comment by smbadiwe** [comment]
@sailfish009 the `1024` feature length is an upper limit. You can surely use something lower. I have used `512` for a project successfully.

If the question is how to reduce? In your data, simply truncate your protein sequences accordingly. For instance, if you want 64, then take the first 64 amino acids and truncate the rest. If the sequence length is less than 64, do nothing.

**Comment by sailfish009** [comment]
If the protein sequence length is 48 then the shape information is [48, 1024]. I want a feature length that is [48, 68].

**Comment by smbadiwe** [comment]
I see. In that case, you'll have to add a linear layer on top of the model. The layer will take 1024 (the dim final layer of the model) as input and 64 as output. Something like this:

```
    class CustomModel(nn.Module):
        def __init__(self):
            super(GenresClassifier, self).__init__()
            self.bert = BertModel.from_pretrained(PRE_TRAINED_MODEL_NAME)
            self.out = nn.Linear(self.bert.config.hidden_size, 64)  # self.bert.config.hidden_size == 1024

        def forward(self, input_ids, attention_mask, token_type_ids):
            _, pooled_output = self.bert(
                input_ids=input_ids,
                attention_mask=attention_mask,
                token_type_ids=token_type_ids
            )
            return self.out(pooled_output)
```

**Comment by sailfish009** [comment]
i think found the answer, sample code is 

`ids = tokenizer.batch_encode_plus(sequences_Example, add_special_tokens=True, max_length=MAX_LENGTH, pad_to_max_length=True, padding='max_length')`
